src/java/org/apache/cassandra/index/sai/disk/v2/V2VectorIndexSearcher.java

```diff
@@ -300,20 +300,9 @@ private CloseableIterator<ScoredRowId> orderByBruteForce(CompressedVectors cv,
                 approximateScores.add(new BruteForceRowIdIterator.RowWithApproximateScore(segmentRowId, ordinal, score));
             }
         }
-        NodeSimilarity.Reranker reranker = o -> similarityFunction.compare(queryVector, getVectorForOrdinal(o));
-        return CloseableIterator.wrap(new BruteForceRowIdIterator(approximateScores, reranker, limit, topK));
-    }
-
-    private float[] getVectorForOrdinal(int ordinal)
-    {
-        try
-        {
-            return graph.getVectorForOrdinal(ordinal);
-        }
-        catch (IOException e)
-        {
-            throw new RuntimeException(e);
-        }
+        var vectorsView = graph.getVectorsView();
+        NodeSimilarity.Reranker reranker = o -> similarityFunction.compare(queryVector, vectorsView.getVectorForOrdinal(o));
```

Comment:
I'm about 60% convinced that we should encapsulate the cleanup in a CloseableReranker class, WDYT?

Reply:
I like that, fixing. I think I'll make the same change to the `GraphSearcher` in `CassandraDiskAnn`

Reply:
I ended up only implementing `CloseableReranker`